### PR TITLE
Update onboarding form validation

### LINKED ISSUE #2584: Conditional continue button reveal
Re how we only reveal the continue button when all inputs have a value...

This concept worked better in prototypes, IRL it's kind of confusing because you don't really know which inputs to fill in before it's revealed.

Let's just keep it permanently visible and use input error states to inform users to fill in required fields:

error

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes #2584

Updates the onboarding form validation to show errors and always show "Continue" button.

@jameskoster I used a method that allows us to easily add custom error strings to each field, so if you have any you want to change, feel free to let me know.  I only used generic messages for the business details step since I was a bit unsure on the language to use there.

### Screenshots
Screen Shot 2019-07-09 at 10 31 33 PM

### Detailed test instructions:

1. Go through the onboarding process.
2. Make sure the "Continue" button is always shown for each step.
3. Make sure validation works for each step and displays errors appropriately.

### CODE REVIEW

File: client/dashboard/profile-wizard/steps/industry.js [hunk 1]
```diff
@@ -80,13 +96,12 @@ class Industry extends Component {
 								/>
 							);
 						} ) }
+						{ error && <span class="woocommerce-profile-wizard__error">{ error }</span> }
```
Comment (timmyc) [inline client/dashboard/profile-wizard/steps/industry.js]: This is throwing a warning in the console for me `s/class/classname`

image

File: client/dashboard/profile-wizard/steps/product-types.js [hunk 2]
```diff
@@ -111,17 +127,16 @@ class ProductTypes extends Component {
 								/>
 							);
 						} ) }
+						{ error && <span class="woocommerce-profile-wizard__error">{ error }</span> }
```
Comment (timmyc) [inline client/dashboard/profile-wizard/steps/product-types.js]: Another spot for the s/class/className

File: client/dashboard/profile-wizard/steps/industry.js (hunk 1 above)
Comment (joshuatf) [reply client/dashboard/profile-wizard/steps/industry.js]: Thanks!  Looks like my linter is on the fritz again.

**Comment by joshuatf** [comment]
>If I skip the required fields, and subsequently select the first ( default ) value, it appears the form field still shows as invalid/error - yet I can proceed to the next step. I would expect the error message to be cleared out once I select a value.

Good catch, Timmy!  We set the default values on focus since the current `SelectControl` doesn't allow a disabled role and would prevent `onChange` from firing when selecting the first item.  Added the validation callback to that function so those errors get removed when the first item is selected.

**Pull request merged**